### PR TITLE
Set up test environment API

### PR DESCRIPTION
**Pull request by FlisAnn**
[Set up test environment API](https://www.pivotaltracker.com/story/show/176031741)
### User Story
```
As a development team,
In order to unit test our code
We would like to set up Rspec
```

## Changes proposed in this pull request:
* Rspec
* FactoryBot
* ShouldaMatcha

## What I have learned working on this feature:
* Setting up a testing framework for the API

## Packages/Gems added
* 'rspec-rails'
* 'shoulda-matchers'
* 'factory_bot_rails'

### CODE REVIEW

File: master.key [hunk 1]
```diff
@@ -0,0 +1 @@
+93ca6e6484a46c2582754f54b5820c49
```
Comment (emtalen) [inline master.key]: 🤯 🤯 🤯 🤯

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+
+ENV['RAILS_ENV'] ||= 'test'
+require File.expand_path('../config/environment', __FILE__)
+
+abort("The Rails environment is running in production mode!") if Rails.env.production?
+require 'rails_helper'
+require 'rspec/rails'
+
```
Comment (emtalen) [inline spec/rails_helper.rb:8]: you need to require the `spec_helper` in here

File: master.key (hunk 1 above)
Comment (emtalen) [reply master.key]: You have to generate a new master key and remove this

**Review by emtalen — CHANGES_REQUESTED** [review]

File: master.key (hunk 1 above)
Comment (tochman) [reply master.key]: 😳 The `master.key` file is added to `.gitignore` as default. Please don't remove stuff from that file that you don't really understand. t's a google search or a question in Slack away.

**Review by emtalen — APPROVED** [review]

**Pull request merged**